static/js/components/SurveyLinkList.jsx

```diff
@@ -1,13 +1,68 @@
 import React from "react";
+import { useSelector } from "react-redux";
 import PropTypes from "prop-types";
-import { ra_to_hours, dec_to_hours } from "../units";
-import styles from "./SurveyLinkList.css";
+import dayjs from "dayjs";
+import utc from "dayjs/plugin/utc";
+import { makeStyles } from "@material-ui/core/styles";
 
-const SurveyLink = ({ name, url }) => (
-  <div className={styles.SurveyLink}>
-    <a href={url}>{name}</a>
-  </div>
-);
+import { ra_to_hours, dec_to_dms } from "../units";
+
+dayjs.extend(utc);
+
+const useStyles = makeStyles(() => ({
+  SurveyLinkList: {
+    display: "inline-block",
+  },
+  SurveyLink: {
+    backgroundColor: "#eee",
+    padding: "0.2em",
+    margin: "0.2em",
+    display: "inline-block",
+  },
+  submitButton: {
+    border: "none",
+    backgroundColor: "transparent",
+    cursor: "pointer",
+    padding: 0,
+    margin: 0,
+    fontWeight: "bold",
+    color: "gray",
+  },
+}));
+
+const SurveyLink = ({ name, url }) => {
+  const styles = useStyles();
+  return (
+    <a href={url} target="new">
+      <div className={styles.SurveyLink}>{name}</div>
+    </a>
+  );
+};
+
+const SurveyLinkForm = ({ name, url, formData, urlEncoded = false }) => {
+  const styles = useStyles();
+  return (
+    <div className={styles.SurveyLink}>
+      <form
+        method="post"
+        target="new"
+        action={url}
+        encType={
+          urlEncoded
+            ? "application/x-www-form-urlencoded"
+            : "multipart/form-data"
+        }
+      >
+        {Object.entries(formData).map(([key, value]) => (
+          <input type="hidden" key={key} name={key} value={value} />
```

Comment:
Nice solution

Reply:
thanks. This took me forever to come up with !